scripts/preinstall.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const { rmSync, readdirSync } = require('fs');
+const { join } = require('path');
+
+const { INIT_CWD, PWD = process.cwd() } = process.env;
+
+// Only run when installed as a dep
+if (!INIT_CWD?.startsWith?.(PWD)) {
+  const depsToKeep = [
```

Comment:
Just curious to know how did we determine this list of deps and know we haven't missed any?

Reply:
Left a comment in the code:
```
  /* These are deps and types which get installed when a production package is installed.
   * When this library is linked as a dep to another project, having all the deps could
   * confuse or conflict the project's compilers.
   *
   * When being installed as dep of another project in production, `node_modules` would
   * be empty.
   */
```